src/otaku.cpp

```diff
@@ -0,0 +1,197 @@
+    return 825000;
+  else
+    return 25000 + (25000 * (1 << (desired_grade - 1)));
+}
+
+bool submersion_cost(struct char_data *ch, bool spend)
+{
+  int desired_grade = GET_GRADE(ch) + 1;
+  long karmacost = (desired_grade + 5) * 300;
+  long nuyencost = calculate_sub_nuyen_cost(desired_grade);
+
+  if (karmacost < 0) {
+    send_to_char("Congratulations, you've hit the ceiling! You can't increase submersion until the code is fixed to allow someone at your rank to do so.\r\n", ch);
+    mudlog_vfprintf(ch, LOG_SYSLOG, "lol init capped at %d for %s, karma overflowed... this game was not designed for that init level", GET_GRADE(ch), GET_CHAR_NAME(ch));
+    return FALSE;
+  }
+
+  long tke = 0;
+  if (karmacost > GET_KARMA(ch) || (GET_KARMA(ch) - karmacost) > GET_KARMA(ch)) {
+    send_to_char(ch, "You do not have enough karma to increase submersion. It will cost you %.2f karma.\r\n", ((float) karmacost / 100));
+    return FALSE;
+  }
+  if (nuyencost > GET_NUYEN(ch) || (GET_NUYEN(ch) - nuyencost) > GET_NUYEN(ch)) {
+    send_to_char(ch, "You do not have enough nuyen to increase submersion. It will cost you %ld nuyen.\r\n", nuyencost);
+    return FALSE;
+  }
+
+  switch (desired_grade) {
+    case 1:
+      tke = 0;
+      break;
+    case 2:
+      tke = 100;
+      break;
+    case 3:
+      tke = 200;
+      break;
+    default:
+      tke = 200 + ((GET_GRADE(ch)-2)*200);
+      break;
+  }
+  if (tke > GET_TKE(ch)) {
+    send_to_char(ch, "You do not have high enough TKE to initiate. You need %ld TKE.\r\n", tke);
+    return FALSE;
+  }
+  if (spend) {
+    lose_nuyen(ch, nuyencost, NUYEN_OUTFLOW_INITIATION);
+    GET_KARMA(ch) -= karmacost;
+  }
+  return TRUE;
+}
+
+void disp_submersion_menu(struct descriptor_data *d)
+{
+  CLS(CH);
+  send_to_char("1) Increase submersion grade\r\n"
+               "2) Learn a new echo\r\n"
+               "3) Return to game\r\n"
+               "Enter initiation option: ", CH);
+  d->edit_mode = SUBMERSION_MAIN;
+}
+
+ACMD(do_submerse)
+{
+  FAILURE_CASE(!IS_OTAKU(ch), "Sorry, submersions are for otaku characters only.");
+
+  skip_spaces(&argument);
+
+  if (subcmd == SCMD_SUBMERSE && submersion_cost(ch, FALSE)) {
+    // Enforce grade restrictions. We can't do this init_cost since it's used elsewhere.
+    if ((GET_GRADE(ch) + 1) > SUBMERSION_CAP) {
+      send_to_char("Congratulations, you've reached the submersion cap! You're not able to advance further.\r\n", ch);
+      return;
+    }
+
+    // Check to see that they can afford it. This sends its own message.
+    if (!submersion_cost(ch, FALSE)) {
+      return;
+    }
+    STATE(ch->desc) = CON_SUBMERSION;
+    PLR_FLAGS(ch).SetBit(PLR_SUBMERSION);
+    disp_submersion_menu(ch->desc);
+    return;
+  }
+}
+
+bool can_select_echo(struct char_data *ch, int i)
+{
+  // Count our grades vs selected echos so we can't learn more than we have grades
+  int available_echoes = GET_GRADE(ch);
+  for (int i = 1; i < ECHO_MAX; i++) {
```

Comment:
Are you intending to shadow `i` here? This makes reading the code a bit confusing, so if we can swap this out to `idx` or something for clarity that'd be appreciated.